<a href="https://colab.research.google.com/github/Ganzer13/data-science-2026/blob/main/Pertemuan_2_Roland_240401010294.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [11]:
import pandas as pd
import numpy as np

# Menggunakan URL raw agar file CSV dapat dibaca langsung oleh pandas
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

# Menampilkan 5 data teratas untuk memastikan data termuat
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [12]:
# Q1: Berapa jumlah baris dan kolom pada dataset titanic? Kolom apa saja yang tersedia?

# Mendapatkan Jumlah baris dan kolom
rows, cols = df.shape
print(f"Jumlah Baris: {rows}")
print(f"Jumlah Kolom: {cols}")

# Menampilkan daftar nama kolom
print("Daftar Kolom:")
print(df.columns.tolist())

Jumlah Baris: 891
Jumlah Kolom: 12
Daftar Kolom:
['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


In [13]:
# Q2: Kolom mana saja yang memiliki missing values? Berapa jumlah dan proporsinya (%) per kolom?

# Menghitung jumlah missing values
missing_count = df.isnull().sum()

# Menghitung proporsi (%)
missing_percentage = (missing_count / len(df)) * 100

# Menggabungkan hasil ke dalam satu DataFrame agar mudah dibaca
missing_data = pd.DataFrame({
    'Jumlah Missing': missing_count,
    'Persentase (%)': missing_percentage
})

print(missing_data[missing_data['Jumlah Missing'] > 0])

          Jumlah Missing  Persentase (%)
Age                  177       19.865320
Cabin                687       77.104377
Embarked               2        0.224467


In [14]:
# Q3: Berapa persen penumpang yang selamat (Survived=1)? Gunakan operasi NumPy atau Pandas.

# Menghitung rata-rata dari kolom Survived (karena 1 = Selamat, 0 = Tidak)
survival_rate = df['Survived'].mean() * 100
print(f"Persentase penumpang selamat: {survival_rate:.2f}%")

Persentase penumpang selamat: 38.38%


In [15]:
# Q4: Filter penumpang wanita dari kelas 1 (Pclass=1). Berapa jumlahnya dan berapa rata-rata usianya?

# Melakukan Filtering
female_class1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]

# Menghitung jumlah dan rata-rata usia
count_female_c1 = len(female_class1)
average_age_female_c1 = female_class1['Age'].mean()

print(f"Jumlah penumpang wanita kelas 1: {count_female_c1}")
print(f"Rata-rata usia mereka: {average_age_female_c1:.2f} tahun")

Jumlah penumpang wanita kelas 1: 94
Rata-rata usia mereka: 34.61 tahun


In [16]:
# Q5: Gunakan groupby untuk menghitung tingkat keselamatan (mean Survived) per kelas penumpang. Kelas mana yang paling tinggi?

# Menggunakan Groupy untuk melihat mean Survived per Pclass
survival_by_class = df.groupby('Pclass')['Survived'].mean() * 100
print(survival_by_class)

Pclass
1    62.962963
2    47.282609
3    24.236253
Name: Survived, dtype: float64


In [17]:
# Q6: Buat analisis tambahan berdasarkan variabel lain yang Anda anggap menarik. Sertakan interpretasi singkat.

# Menghitung tingkat keselamatan berdasarkan jenis kelamin
survival_by_gender = df.groupby('Sex')['Survived'].mean() * 100
print(survival_by_gender)

Sex
female    74.203822
male      18.890815
Name: Survived, dtype: float64


Temuan: Terdapat perbedaan yang sangat kontras; wanita memiliki tingkat keselamatan sekitar 74%, sedangkan pria hanya sekitar 18%. Hal ini memperkuat teori kebijakan "wanita dan anak-anak didahulukan" saat evakuasi. Selain itu, menarik untuk melihat korelasi antara tarif tiket (Fare) dengan keselamatan, di mana biasanya tarif yang lebih mahal memberikan akses evakuasi yang lebih cepat.